packages/statemanager/src/statefulBinaryTreeStateManager.ts

```diff
@@ -0,0 +1,802 @@
+
+      let lastChunkByteIndex = BINARY_TREE_CODE_CHUNK_SIZE
+      // Determine code ending byte (if we're on the last chunk)
+      if (x === chunks.length - 1) {
+        // On the last chunk, the slice either ends on a partial chunk (if codeSize doesn't exactly fit in full chunks), or a full chunk
+        lastChunkByteIndex = codeSize % BINARY_TREE_CODE_CHUNK_SIZE || BINARY_TREE_CODE_CHUNK_SIZE
+      }
+      code.set(
+        chunks[x]!.slice(1, lastChunkByteIndex + 1),
+        code.byteOffset + x * BINARY_TREE_CODE_CHUNK_SIZE,
+      )
+    }
+    this._caches?.code?.put(address, code)
+
+    return code
+  }
+
+  getCodeSize = async (address: Address): Promise<number> => {
+    const accountBytes = (
+      await this._tree.get(getBinaryTreeStem(this.hashFunction, address), [
+        BinaryTreeLeafType.BasicData,
+      ])
+    )[0]
+    if (accountBytes === null) return 0
+    return decodeBinaryTreeLeafBasicData(accountBytes).codeSize
+  }
+  getStorage = async (address: Address, key: Uint8Array): Promise<Uint8Array> => {
+    if (key.length !== 32) {
+      throw EthereumJSErrorWithoutCode('Storage key must be 32 bytes long')
+    }
+    const cachedValue = this._caches?.storage?.get(address, key)
+    if (cachedValue !== undefined) {
+      const decoded = RLP.decode(cachedValue ?? new Uint8Array(0)) as Uint8Array
+      return decoded
+    }
+
+    const account = await this.getAccount(address)
+    if (!account) {
+      return new Uint8Array()
+    }
+    const storageKey = getBinaryTreeKeyForStorageSlot(
+      address,
+      bytesToBigInt(key),
+      this.hashFunction,
+    )
+    const value = await this._tree.get(storageKey.slice(0, 31), [storageKey[31]])
+
+    this._caches?.storage?.put(address, key, value[0] ?? hexToBytes('0x80'))
+    const decoded = (value[0] ?? new Uint8Array(0)) as Uint8Array
+    return setLengthLeft(decoded, 32)
+  }
+
+  putStorage = async (address: Address, key: Uint8Array, value: Uint8Array): Promise<void> => {
+    this._caches?.storage?.put(address, key, RLP.encode(value))
+    if (this._caches?.storage === undefined) {
+      const storageKey = getBinaryTreeKeyForStorageSlot(
+        address,
+        bytesToBigInt(key),
+        this.hashFunction,
+      )
+      await this._tree.put(storageKey.slice(0, 31), [storageKey[31]], [setLengthLeft(value, 32)])
+    }
+  }
+
+  clearStorage = async (address: Address): Promise<void> => {
+    // TODO: Determine if it's possible to clear the actual slots in the trie
+    // since the EIP doesn't seem to state how to handle this
+    // The main concern I have is that we have no way of identifying all storage slots
+    // for a given account so we can't correctly update the trie's root hash
+    // (since presumably "clearStorage" would imply writing over all of the storage slots with zeros)
+    // Also, do we still need a storageRoot? - presumably not since we don't have separate storage tries
+    this._caches?.storage?.clearStorage(address)
```

Comment:
```suggestion
    this._caches?.storage?.clearStorage(address)
```

As above, I think we can remove this comment.  We don't have any concept in production code for "clearing storage" aside from maybe account SELFDESTRUCT and I don't know what that looks like in a unifed state tree world yet.  Should wait to worry about code/storage stems associated with an account until we have clear decisions about even allowing this sort of thing to happen (e.g. where we wipe an account's storage or code)